pkg/catalogv2/git/utils.go

```diff
@@ -0,0 +1,85 @@
+package git
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	stateDir             = "management-state/git-repo"
+	staticDir            = "/var/lib/rancher-data/local-catalogs/v2"
+	localDir             = "../rancher-data/local-catalogs/v2" // identical to helm.InternalCatalog
+	localReferenceBranch = "refs/heads/"
+)
+
+func gitDir(namespace, name, gitURL string) string {
+	staticDir := filepath.Join(staticDir, namespace, name, hash(gitURL))
+	if s, err := os.Stat(staticDir); err == nil && s.IsDir() {
+		return staticDir
+	}
+	localDir := filepath.Join(localDir, namespace, name, hash(gitURL))
+	if s, err := os.Stat(localDir); err == nil && s.IsDir() {
+		return localDir
+	}
+	return filepath.Join(stateDir, namespace, name, hash(gitURL))
+}
+
+// isBundled check if the repositories are bundled on local static directory
+func isBundled(git *git) bool {
+	return strings.HasPrefix(git.Directory, staticDir)
```

Comment:
This seems out of dated. Copy the new function.

Reply:
I have implemented a fix for this, please resolve the conversation